src/vcf/record.cpp

```diff
@@ -344,33 +344,33 @@ namespace ebi
     void Record::strict_validation_info_predefined_tags(std::string const & field_key, std::string const & field_value,
                                                         std::vector<std::string> const & values) const
     {
-        if (field_key == "AA") {
+        if (field_key == AA) {
             static boost::regex aa_regex("((?![,;=])[[:print:]])+");
             if (!boost::regex_match(field_value, aa_regex)) {
                 throw new InfoBodyError{line, "INFO AA=" + field_value + " value is not a single dot or a string of bases", field_key};
             }
-        } else if (field_key == "AF") {
+        } else if (field_key == AF) {
             for (auto & value : values) {
                 if (std::stold(value) < 0 || std::stold(value) > 1) {
                     throw new InfoBodyError{line, "INFO AF=" + field_value + " value does not lie in the interval [0,1]", field_key};
                 }
             }
-        } else if (field_key == "CIGAR") {
+        } else if (field_key == CIGAR) {
             static boost::regex cigar_string("([0-9]+[MIDNSHPX])+");
             for (auto & value : values) {
                 if (!boost::regex_match(value, cigar_string)) {
                     throw new InfoBodyError{line, "INFO CIGAR=" + field_value + " value is not an alphanumeric string compliant with the SAM specification", field_key};
                 }
             }
-        } else if (field_key == "END") {
-            auto it = info.find("IMPRECISE");
+        } else if (field_key == END) {
+            auto it = info.find(IMPRECISE);
             if (it != info.end() && it->second == "0") {
                 auto expected = std::to_string(position + reference_allele.length() - 1);
                 if (field_value != expected) {
                     throw new InfoBodyError{line, "INFO END=" + field_value + " value must be equal to \"POS + length of REF - 1\" for a precise variant (where IMPRECISE is set to 0), expected " + expected, field_key};
                 }
             }
-        } else if (field_key == "SVLEN" && values.size() == alternate_alleles.size()) {
+        } else if (field_key == SVLEN && values.size() == alternate_alleles.size()) {
             bool is_ref_symbolic = true;
```

Comment:
this is used sometimes properly and sometimes as `is_ref_not_symbolic`. later in this function, the code under `if (is_ref_symbolic && is_alt_symbolic[i])` is in fact what we should do if the alleles are not symbolic, which is taking the difference in allele string sizes. also, the reference can not be a symbolic allele (the ragel code checks that under the record_ref machine)